src/build-remote/build-remote.cc

```diff
@@ -123,11 +130,18 @@ static int _main(int argc, char * * argv)
                 for (auto & m : machines) {
                     debug("considering building on remote machine '%s'", m.storeUri);
 
-                    if (m.enabled && std::find(m.systemTypes.begin(),
-                            m.systemTypes.end(),
-                            neededSystem) != m.systemTypes.end() &&
-                        m.allSupported(requiredFeatures) &&
-                        m.mandatoryMet(requiredFeatures)) {
+                    if (!m.enabled) {
+                        debug("declined remote machine '%s' because it is disabled", storeUri);
+                    } else if (std::find(m.systemTypes.begin(), m.systemTypes.end(), neededSystem)
+                        == m.systemTypes.end()) {
+                        printInfo(format("declined remote machine '%1%' for building '%2%' because it does not have the needed system type: '%3%") % m.storeUri % drvstr % neededSystem);
```

Comment:
```suggestion
                        printInfo("declined remote machine '%s' for building '%s' because it does not have the needed system type '%s", m.storeUri, drvstr, neededSystem);
```

`format` is deprecated.

Reply:
> `format` is deprecated.

Ah OK, I somehow thought it was the other way around.